src/jit/gentree.cpp

```diff
@@ -7912,7 +7925,7 @@ GenTreePtr Compiler::gtCloneExpr(
                 copy                              = new (this, GT_HWIntrinsic)
                     GenTreeHWIntrinsic(hwintrinsicOp->TypeGet(), hwintrinsicOp->gtGetOp1(),
                                        hwintrinsicOp->gtGetOp2IfPresent(), hwintrinsicOp->gtHWIntrinsicId,
-                                       hwintrinsicOp->gtSIMDBaseType, hwintrinsicOp->gtSIMDSize);
+                                       hwintrinsicOp->gtSIMDBaseType, hwintrinsicOp->gtSIMDSize, hwintrinsicOp->gtNumArgs);
```

Comment:
Do we still need to carry this information here? It seems like something well suited for the table (or being implicit from the category)

Reply:
Intrinsics may have overloads with different parameter number.

Reply:
Those are limited to the `Special` intrinsics though, right?

Most of the intrinsics look to be 2, with a few at 1, 0, or 3. Those above 3 are very limited (and mostly look to be the `Set` intrinsics).

Reply:
Yes.

Reply:
It seems like, since we will have to have special handling for those anyways (and since that handling can likely be limited to the importer), it isn't worthwhile to carry around the arg count.

I'm not sure how useful it will be in CodeGen either, since two binary intrinsics may have completely different handling depending on supported encodings.

Reply:
Agree. Will remove.